src/api/FactoryAPI.ts

```diff
@@ -138,6 +144,78 @@ export class FactoryAPI {
     return new BigNumber(10 ** (18 - minDecimal.toNumber()));
   }
 
+  public async calculateRequiredComponentUnits(
+    componentPrices: BigNumber[],
+    components: Address[],
+    componentProportions: BigNumber[],
+    targetSetPrice: BigNumber,
+  ): Promise<BigNumber[]> {
+    this.assert.common.proportionsSumToOne(componentProportions, coreAPIErrors.PROPORTIONS_DONT_ADD_UP_TO_1());
```

Comment:
Conforming to the rest of the API, can you move all of these assertions into a method called `assertCalculateRequiredComponentUnits`

Reply:
+1